.azure/pipelines/jobs/default-build.yml

```diff
@@ -78,151 +78,357 @@ parameters:
+          displayName: Disk utilization
+      - ${{ if eq(parameters.agentOs, 'macOS') }}:
+        - script: df -h && du -h -d 3 ..
+          displayName: Disk utilization
+
+      - ${{ if and(eq(parameters.agentOs, 'Windows'), eq(parameters.isAzDOTestingJob, true)) }}:
+        - powershell: ./eng/scripts/FinishDumpCollectionForHangingBuilds.ps1 artifacts/dumps/
+          displayName: Finish background dump collection
+          continueOnError: true
+          condition: always()
+
+      - ${{ if eq(parameters.agentOs, 'Windows') }}:
+        - powershell: eng\scripts\KillProcesses.ps1
+          displayName: Kill processes
+          continueOnError: true
+          condition: always()
+      - ${{ if ne(parameters.agentOs, 'Windows') }}:
+        - script: eng/scripts/KillProcesses.sh
+          displayName: Kill processes
+          continueOnError: true
+          condition: always()
+
+      - ${{ each artifact in parameters.artifacts }}:
+        - task: PublishBuildArtifacts@1
+          displayName: Upload artifacts from ${{ artifact.path }}
+          condition: and(or(succeeded(), eq('${{ artifact.publishOnError }}', 'true')), or(eq(variables['system.pullrequest.isfork'], false), eq('${{ artifact.includeForks }}', 'true')))
+          continueOnError: true
+          inputs:
+            # Assume runtime variable values are absolute paths already.
+            ${{ if startsWith(artifact.path, '$(') }}:
+              pathToPublish: ${{ artifact.path }}
+            ${{ if not(startsWith(artifact.path, '$(')) }}:
+              pathToPublish: $(Build.SourcesDirectory)/${{ artifact.path }}
+            ${{ if eq(artifact.name, '') }}:
+              artifactName: artifacts-$(AgentOsName)-$(BuildConfiguration)
+            ${{ if ne(artifact.name, '') }}:
+              artifactName: ${{ artifact.name }}
+            artifactType: Container
+            parallel: true
+
+      - ${{ if eq(parameters.agentOs, 'Windows') }}:
+        - powershell: $(Build.SourcesDirectory)/eng/scripts/UploadCores.ps1 -ProcDumpOutputPath artifacts/dumps/
+          condition: failed()
+          displayName: Upload cores
+      - ${{ if ne(parameters.agentOs, 'Windows') }}:
+        - script: $(Build.SourcesDirectory)/eng/scripts/upload-cores.sh
+          condition: failed()
+          displayName: Upload cores
+
+      - ${{ if and(eq(parameters.isAzDOTestingJob, true), ne(parameters.enablePublishTestResults, false)) }}:
+        - task: PublishTestResults@2
+          displayName: Publish js test results
+          condition: always()
+          inputs:
+            testResultsFormat: JUnit
+            testResultsFiles: '**/artifacts/log/**/*.junit.xml'
+            testRunTitle: $(AgentOsName)-$(BuildConfiguration)-js
+            mergeTestResults: true
+            buildConfiguration: $(BuildConfiguration)
+            buildPlatform: $(AgentOsName)
+
+- ${{ if eq(variables['System.TeamProject'], 'internal') }}:
```

Comment:
As discussed offline, I would split these steps into own file for the ease of maintenance. I'd name these files something like build-public.yml and buid-internal.yml.